database/config.go

```diff
@@ -0,0 +1,108 @@
+package database
+
+import (
+	"time"
+
+	"github.com/flyteorg/flytestdlib/config"
+	"gorm.io/gorm/logger"
+)
+
+const (
+	database = "database"
+	postgres = "postgres"
+)
+
+//go:generate pflags DbConfig --default-var=defaultConfig
+
+var defaultConfig = &DbConfig{
+	MaxIdleConnections: 10,
+	MaxOpenConnections: 1000,
+	ConnMaxLifeTime:    config.Duration{Duration: time.Hour},
+	Postgres: &PostgresConfig{
+		Port:         5432,
+		User:         postgres,
+		Host:         postgres,
+		DbName:       postgres,
+		ExtraOptions: "sslmode=disable",
+	},
+}
+var configSection = config.MustRegisterSection(database, defaultConfig)
+
+// DbConfig is used to for initiating the database connection with the store that holds registered
+// entities (e.g. workflows, tasks, launch plans...)
+type DbConfig struct {
+	// deprecated: Please use Postgres.Host
+	DeprecatedHost string `json:"host" pflag:",deprecated"`
+	// deprecated: Please use Postgres.Port
+	DeprecatedPort int `json:"port" pflag:",deprecated"`
+	// deprecated: Please use Postgres.DbName
+	DeprecatedDbName string `json:"dbname" pflag:",deprecated"`
+	// deprecated: Please use Postgres.User
+	DeprecatedUser string `json:"username" pflag:",deprecated"`
+	// deprecated: Please use Postgres.Password
+	DeprecatedPassword string `json:"password" pflag:",deprecated"`
+	// deprecated: Please use Postgres.PasswordPath
+	DeprecatedPasswordPath string `json:"passwordPath" pflag:",deprecated"`
+	// deprecated: Please use Postgres.ExtraOptions
+	DeprecatedExtraOptions string `json:"options" pflag:",deprecated"`
+	// deprecated: Please use Postgres.Debug
+	DeprecatedDebug bool `json:"debug" pflag:",deprecated"`
+
+	EnableForeignKeyConstraintWhenMigrating bool            `json:"enableForeignKeyConstraintWhenMigrating" pflag:",Whether to enable gorm foreign keys when migrating the db"`
+	MaxIdleConnections                      int             `json:"maxIdleConnections" pflag:",maxIdleConnections sets the maximum number of connections in the idle connection pool."`
+	MaxOpenConnections                      int             `json:"maxOpenConnections" pflag:",maxOpenConnections sets the maximum number of open connections to the database."`
+	ConnMaxLifeTime                         config.Duration `json:"connMaxLifeTime" pflag:",sets the maximum amount of time a connection may be reused"`
+	LogLevel                                logger.LogLevel `json:"log_level" pflag:"-,"`
+	Postgres                                *PostgresConfig `json:"postgres,omitempty"`
+	SQLite                                  *SQLiteConfig   `json:"sqlite,omitempty"`
```

Comment:
I would say let's keep these non-pointers... they lead to nasty nil refs...
```suggestion
	Postgres                               PostgresConfig `json:"postgres,omitempty"`
	SQLite                                  SQLiteConfig   `json:"sqlite,omitempty"`
```